firebase-perf/src/test/java/com/google/firebase/perf/application/FragmentStateMonitorTest.java

```diff
@@ -88,197 +92,176 @@ public void setUp() {
     doAnswer((Answer<Timer>) invocationOnMock -> new Timer(currentTime)).when(clock).getTime();
 
     DeviceCacheManager.clearInstance();
-    ConfigResolver.clearInstance();
-
-    ConfigResolver configResolver = ConfigResolver.getInstance();
-    configResolver.setDeviceCacheManager(new DeviceCacheManager(new FakeDirectExecutorService()));
-    ConfigResolver spyConfigResolver = spy(configResolver);
-    doReturn(true).when(spyConfigResolver).isPerformanceMonitoringEnabled();
-    this.configResolver = spyConfigResolver;
-
-    // fmaMetrics1 should have 1+3+1=5 total frames, 3+1=4 slow frames, and 1 frozen frames.
-    SparseIntArray sparseIntArray = new SparseIntArray();
-    sparseIntArray.append(1, 1);
-    sparseIntArray.append(17, 3);
-    sparseIntArray.append(800, 1);
-    fmaMetrics1[FrameMetricsAggregator.TOTAL_INDEX] = sparseIntArray;
-
-    // fmaMetrics2 should have 5+5+4=14 total frames, 5+4=9 slow frames, and 4 frozen frames.
-    sparseIntArray = new SparseIntArray();
-    sparseIntArray.append(1, 5);
-    sparseIntArray.append(18, 5);
-    sparseIntArray.append(800, 4);
-    fmaMetrics2[FrameMetricsAggregator.TOTAL_INDEX] = sparseIntArray;
+    when(configResolver.isPerformanceMonitoringEnabled()).thenReturn(true);
+
+    doNothing().when(recorder).start();
+    doNothing().when(recorder).startFragment(any());
+
+    activity = createFakeActivity(true);
+
+    // Sample frame counts.
+    frameCounts1 = new PerfFrameMetrics(9, 5, 3);
+    frameCounts2 = new PerfFrameMetrics(14, 9, 4);
+
+    savedInstanceState = mock(Bundle.class);
+    appStateMonitor = mock(AppStateMonitor.class);
+    mockFragment1 = mock(Fragment.class);
+    mockFragment2 = mock(Fragment.class);
   }
 
   /************ Trace Creation Tests ****************/
 
   @Test
-  public void lifecycleCallbacks_differentFrameMetricsCapturedByFma_logFragmentScreenTrace() {
+  public void fragmentTraceName_validFragment_validFragmentScreenTraceNameGenerated() {
     FragmentStateMonitor monitor =
-        new FragmentStateMonitor(clock, mockTransportManager, appStateMonitor, fma);
-    when(fma.getMetrics()).thenReturn(fmaMetrics1);
+        new FragmentStateMonitor(clock, mockTransportManager, appStateMonitor, recorder);
+    Fragment testFragment = new FragmentMonitorMockFragment();
+    Assert.assertEquals(
```

Comment:
Can we use the truth framework? (assertThat...)

I'd recommend using it consistently whenever we touch a test. That way eventually all / most tests will be using it.

Reply:
Good idea. Moved out all such asserts to use truth framework.